ios/RNCCameraRollManager.m

```diff
@@ -98,62 +98,96 @@ static void requestPhotoLibraryAccess(RCTPromiseRejectBlock reject, PhotosAuthor
+        assetRequest = [PHAssetChangeRequest creationRequestForAssetFromImage:inputImage];
       }
-
-      placeholder = [changeRequest placeholderForCreatedAsset];
-    } completionHandler:^(BOOL success, NSError * _Nullable error) {
-      if (success) {
+      placeholder = [assetRequest placeholderForCreatedAsset];
+      if(![options[@"album"] isEqualToString:@""]){
+        photosAsset = [PHAsset fetchAssetsInAssetCollection:collection options:nil];
+        PHAssetCollectionChangeRequest *albumChangeRequest = [PHAssetCollectionChangeRequest changeRequestForAssetCollection:collection assets:photosAsset];
+        [albumChangeRequest addAssets:@[placeholder]];
+      }
+    } completionHandler:^(BOOL success, NSError *error) {
+      if (success){
         NSString *uri = [NSString stringWithFormat:@"ph://%@", [placeholder localIdentifier]];
         resolve(uri);
-      } else {
+      }else{
         reject(kErrorUnableToSave, nil, error);
       }
     }];
   };
-
+  void (^saveWithOptions)(void) = ^void() {
+    if (![options[@"album"] isEqualToString:@""]) {
+      
+      PHFetchOptions *fetchOptions = [[PHFetchOptions alloc] init];
+      fetchOptions.predicate = [NSPredicate predicateWithFormat:@"title = %@", options[@"album"] ];
+      collection = [PHAssetCollection fetchAssetCollectionsWithType:PHAssetCollectionTypeAlbum
+                                                            subtype:PHAssetCollectionSubtypeAny
+                                                            options:fetchOptions].firstObject;
+      // Create the album
+      if (!collection){
+        [[PHPhotoLibrary sharedPhotoLibrary] performChanges:^{
+          PHAssetCollectionChangeRequest *createAlbum = [PHAssetCollectionChangeRequest creationRequestForAssetCollectionWithTitle:options[@"album"]];
+          placeholder = [createAlbum placeholderForCreatedAssetCollection];
+        } completionHandler:^(BOOL success, NSError *error) {
+          if (success){
+            PHFetchResult *collectionFetchResult = [PHAssetCollection fetchAssetCollectionsWithLocalIdentifiers:@[placeholder.localIdentifier]
+                                                                                                        options:nil];
+            collection = collectionFetchResult.firstObject;
+            saveBlock();
+          }else{
+            reject(kErrorUnableToSave, nil, error);
+          }
+        }];
+      }else{
+        saveBlock();
+      }
+    }else{
+      saveBlock();
+    }
+  };
+  
   void (^loadBlock)(void) = ^void() {
-    if ([type isEqualToString:@"video"]) {
+    if ([options[@"type"] isEqualToString:@"video"]) {
       inputURI = request.URL;
-      saveBlock();
+      saveWithOptions();
     } else {
       [self.bridge.imageLoader loadImageWithURLRequest:request callback:^(NSError *error, UIImage *image) {
         if (error) {
           reject(kErrorUnableToLoad, nil, error);
           return;
         }
-
+        
```

Comment:
don't add spaces on empty lines

Reply:
Maybe we should provide formatter configs with EditorConfig or specific configs for AndroidStudio and XCode like it's done with prettier. I used the XCode formatter previosly, which also inserts the whitespaces. When i use AndroidStudio default, git blame would become useless.